python/ray/data/_internal/stats.py

```diff
@@ -793,6 +793,29 @@ def to_string(
                 out += "\nDataset memory:\n"
                 out += "* Spilled to disk: {}MB\n".format(dataset_mb_spilled)
 
+            output_num_rows = self.operators_stats[-1].output_num_rows
+            total_num_out_rows = output_num_rows["sum"] if output_num_rows else 0
+            total_wall_time = sum(
+                [
+                    op_stats.wall_time["sum"]
+                    for op_stats in self.operators_stats
+                    if op_stats.wall_time
+                ]
+            )
+            if total_num_out_rows and self.time_total_s and total_wall_time:
+                out += "\n"
+                out += "Dataset throughput:\n"
+                out += (
+                    "    * Ray Data throughput:"
+                    f" {total_num_out_rows / self.time_total_s} "
+                    "rows/s\n"
+                )
+                out += (
+                    "    * Optimal single node throughput:"
```

Comment:
thought about it some more, i think "optimal" could be confusing to users. thoughts on using "Estimated single node throughput" here and in repr?

Reply:
Yeah I think that makes more sense, will update.